### PR TITLE
Added save and update functionality to app

### PR DESCRIPTION
**Pull request by Y0z64**
- **Fixed UI and color issues**
- **Migrated db**
- **Added update and delete functions**
- **Added saving and updating**
- **Added manual save button to toolbar**
- **Made it, so the page tab title is the note title instead of the ID**

**Added auto-save functionality**: After every change to the markdown content it starts a countdown that is reset for every other change made during the countdown, finally, if it reaches 0 it will automatically save

Save button functions as an indicator that there is a save in progress, this indicator activates even if the save is initiated by the auto save functionality

## Issues:
- Tried to add a way to save when user changed route, but saving when the component unmounted caused a ddos attack to the rack API thanks to astro shenanigans
- Updating required deleting previous entry in rack and uploading a new one, it takes around ~13 to do the request which is not optimal and should be fixed

### CODE REVIEW

**Review by Gorchon — COMMENTED** [review]
karamba

File: src/components/react/MKEditor.tsx [hunk 1]
```diff
@@ -94,11 +129,11 @@ export default function App() {
...
           onChange={(newMarkdown) => {
-            setMarkdown(newMarkdown);
+            setContent(newMarkdown);
             onChange?.(newMarkdown);
           }}
-          markdown={markdown}
-          className="outline-orange-400 outline w-full "
+          markdown={content}
+          className="w-full "
           contentEditableClassName="prose dark:text-white"
```
Comment (DanielBrMz) [inline src/components/react/MKEditor.tsx]: ¿Es posible cambiar el color del texto que se renderiza cuando escribes títulos?

Por ejemplo: 
`# Titulo 1` -> 
# Titulo 1 

El Preview Text sigue apareciendo en color negro cuando el modo oscuro está activo

**Review by DanielBrMz — APPROVED** [review]
Algo bonito, si puedes encontrar la manera de cambiar la configuración acerca del cometario que hice estaría atm

File: src/components/react/MKEditor.tsx (hunk 1 above)
Comment (Y0z64) [reply src/components/react/MKEditor.tsx]: Yeah

Comment (Y0z64) [reply src/components/react/MKEditor.tsx:158]: Arregladisimo padre

**Comment by Y0z64** [comment]
Mergeando sin aprovacion del @adpadillar como todo un chico peligroso que le gusta el riesgo

**Pull request merged**